### PR TITLE
EVG-2934: string or bool for s3put optional

### PR DESCRIPTION
**Pull request by tychoish**

### CODE REVIEW

File: command/s3_put.go [hunk 1]
```diff
@@ -71,11 +71,12 @@ type s3put struct {
 	// Optional, when set to true, causes this command to be skipped over without an error when
 	// the path specified in local_file does not exist. Defaults to false, which triggers errors
 	// for missing files.
-	Optional bool `mapstructure:"optional"`
+	Optional util.StringOrBool `mapstructure:"optional" plugin:"expand"`
```
Comment (john-m-liu) [inline command/s3_put.go]: What does the plugin tag do/where is it used?

Comment (tychoish) [reply command/s3_put.go]: it's the thing that makes expansions work

**Comment by tychoish** [comment]
I added a few cases to the exist test to make sure that expansions actually work. (it looks like they do.)

File: command/s3_put_test.go [hunk 2]
```diff
@@ -206,52 +221,70 @@ func TestS3PutValidateParams(t *testing.T) {
...
+
+			conf.Expansions.Update(
+				map[string]string{
+					"aws_key":      "key",
+					"aws_secret":   "secret",
+					"remote_file":  "remote",
+					"bucket":       "bck",
+					"content_type": "ct",
+					"display_name": "file",
+					"visibility":   artifact.Private,
```
Comment (bsamek) [inline command/s3_put_test.go]: Did you also want to test optional?

File: util/type_string_or_bool.go [hunk 3]
```diff
@@ -0,0 +1,86 @@
+package util
```
Comment (bsamek) [inline util/type_string_or_bool.go]: Add tests for marshal/unmarshal yaml/json

**Pull request merged**